source/azure_iot_provisioning_client.c

```diff
@@ -0,0 +1,963 @@
+                                                                       ( char * ) xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer,
+                                                                       azureiotTOPIC_MAX, &xMQTTTopicLength );
+        }
+        else
+        {
+            xCoreResult =
+                az_iot_provisioning_client_query_status_get_publish_topic( &xAzureIoTProvisioningClientHandle->_internal.xProvisioningClientCore,
+                                                                           xAzureIoTProvisioningClientHandle->_internal.xRegisterResponse.operation_id,
+                                                                           ( char * ) xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer,
+                                                                           azureiotTOPIC_MAX, &xMQTTTopicLength );
+        }
+
+        if( az_result_failed( xCoreResult ) )
+        {
+            prvAzureIoTProvisioningClientUpdateState( xAzureIoTProvisioningClientHandle, eAzureIoTProvisioningFailed );
+            return;
+        }
+
+        if( ( xReturn =
+                  prvAzureIoTProvisioningClientRequestPayload( xAzureIoTProvisioningClientHandle,
+                                                               xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer +
+                                                               xMQTTTopicLength,
+                                                               xAzureIoTProvisioningClientHandle->_internal.ulProvisioningScratchBufferLength -
+                                                               ( uint32_t ) xMQTTTopicLength,
+                                                               &xMQTTPayloadLength ) ) != eAzureIoTProvisioningSuccess )
+        {
+            prvAzureIoTProvisioningClientUpdateState( xAzureIoTProvisioningClientHandle, xReturn );
+            return;
+        }
+
+        ( void ) memset( ( void * ) &xMQTTPublishInfo, 0x00, sizeof( xMQTTPublishInfo ) );
+        xMQTTPublishInfo.xQOS = eAzureIoTMQTTQoS0;
+        xMQTTPublishInfo.pcTopicName = xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer;
+        xMQTTPublishInfo.usTopicNameLength = ( uint16_t ) xMQTTTopicLength;
+        xMQTTPublishInfo.pvPayload = xAzureIoTProvisioningClientHandle->_internal.pucProvisioningScratchBuffer + xMQTTTopicLength;
+        xMQTTPublishInfo.xPayloadLength = xMQTTPayloadLength;
+        usPublishPacketIdentifier = AzureIoTMQTT_GetPacketId( &( xAzureIoTProvisioningClientHandle->_internal.xMQTTContext ) );
+
+        if( ( xResult = AzureIoTMQTT_Publish( &( xAzureIoTProvisioningClientHandle->_internal.xMQTTContext ),
+                                              &xMQTTPublishInfo, usPublishPacketIdentifier ) ) != eAzureIoTMQTTSuccess )
+        {
+            AZLogError( ( "Failed to publish. Error=%d", xResult ) );
```

Comment:
Nit: Consider only a few extra words here, like `Failed to publish prov request` or something.  My concern would be if we end up with error logs at some point without __FUNCTION__ / __LINE__ / __FILE__ then there's change this could be somewhat ambiguous if there's >1 "failed to publish."

We may be able to piece together where the line we cared about was from context/other log messages, but doing a straight search through src for exact msg is easier.